### PR TITLE
[SCHEMA] Tabular metadata rules for both top and second-level tabular files

### PR DESCRIPTION
**Pull request by TheChymera**
* Representation of second-level files added.
* Homogenization of modality and non-modality file representation.
* Documenting file identifiers such as README and _sessions as
  suffixes..

Co-authored-by: Taylor Salo <tsalo006@fiu.edu>

----

As per https://github.com/bids-standard/bids-specification/pull/987#issuecomment-1027163553

@tsalo @yarikoptic

### CODE REVIEW

**Comment by TheChymera** [comment]
Test failure seems to be some website which isn't available and was not introduced by this PR. Might even be temporary, but apparently I don't have permission to re-run the tests.

File: src/schema/objects/suffixes.yaml [hunk 1]
```diff
@@ -482,6 +505,10 @@ coordsystem:
   description: |
     A JSON document specifying the coordinate system(s) used for the MEG, EEG,
     head localization coils, and anatomical landmarks.
+dataset_description:
```
Comment (tsalo) [inline src/schema/objects/suffixes.yaml:508]:
```suggestion
# This is a special top-level file, which is why an underscore is allowed in the "suffix".
dataset_description:
```

I'm thinking we could just add comments to this one and `genetic_info` so folks don't think they can include underscores in new suffixes.

Comment (TheChymera) [reply src/schema/objects/suffixes.yaml:508]: Yeah, I really liked this idea, but as per the prevalent opinion in the discussion today apparently we should just strike it down as a suffix :(

Maybe we can bring the suffixification discussion back later after we get a foot in the door with this ^^

Comment (tsalo) [reply src/schema/objects/suffixes.yaml:508]: Agreed!

**Comment by TheChymera** [comment]
@tsalo @yarikoptic

I have incorporated the amendments from last week's meeting which reduced the scope of “suffixification”. This seems like a fairly simple self-contained addition of files previously not covered by the YAML which closes https://github.com/bids-standard/bids-specification/issues/978 .

So maybe we could discuss whether this can just be pulled today?

One small addition which I'd like to mention is that now we have temporarily given up on homogenizing the naming specification for all files.
Thus we have just created a representation for 2 new files which behave analogously to `rules/datatypes/*.yaml` without changing the representations of other non-modality files.
Perhaps it would be better, after all, to put the new `tabular_metadata.yaml` file in `rules/datatypes/tabular_metadata.yaml`, as [previously suggested](https://github.com/bids-standard/bids-specification/pull/985#issuecomment-1019573787)?
This would also make parsing of the schema into regex or placeholder example paths easier and compatible with current functions.

**Comment by effigies** [comment]
FWIW I approve of moving participants and samples into this group, but also okay with postponing it.

**Review by effigies — APPROVED** [review]

**Review by tsalo — APPROVED** [review]
LGTM. Thanks!

File: src/schema/rules/tabular_metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+---
```
Comment (yarikoptic) [inline src/schema/rules/tabular_metadata.yaml:1]: @effigies  we already have

```
❯ grep -A12 -E '^(scans|sessions)' src/schema/rules/files/common/tables.yaml
scans:
  level: optional
  suffixes:
    - scans
  extensions:
    - .tsv
    - .json
  entities:
    subject: required
    session: optional # session is required if session is present in the dataset.
sessions: # This file may only exist if session is present in the dataset.
  level: optional
  suffixes:
    - sessions
  extensions:
    - .tsv
    - .json
  entities:
    subject: required
```

seemingly with identical content (just dict not list).

How this file is used "in addition"?

Comment (yarikoptic) [reply src/schema/rules/tabular_metadata.yaml:1]: Sorry for the noise @effigies ! my bad -- how the hack I got to review this ? ;)    codecov woke up 2 years later and I thought that this PR got just merged, so I got all confused

**Pull request merged**